server/lib/save_data_and_file.coffee

```diff
@@ -20,6 +20,11 @@ module.exports = (log, model, options, tags) ->
 
         # For each entry...
         async.eachSeries entriesToSave, (entry, callback) ->
+            if (entry.date not instanceof moment)
+                log.error 'Moment instance expected for date field'
+                log.info 'Bill creation aborted'
+                return callback()
```

Comment:
Seems to be a good compromise. Watching more carefully the rest of the code, the errors are never shown to the user, you can keep it like that.